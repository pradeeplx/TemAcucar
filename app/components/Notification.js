import React, { Component } from 'react'
import { TouchableOpacity, View, StyleSheet } from 'react-native'
import Colors from "../Colors"
import HtmlSentence from "./HtmlSentence"
import TimeAgo from "./TimeAgo"

export default class Notification extends Component {
  handlePress() {
    const { onView, notification } = this.props
    onView && onView(notification)
  }

  render() {
    const { read, text, triggering_user, created_at } = this.props.notification
    return (
      <TouchableOpacity
        onPress={this.handlePress.bind(this)}
        style={{
          backgroundColor: (read ? Colors.white : Colors.lightBeige),
          borderColor: Colors.gray,
          borderBottomWidth: StyleSheet.hairlineWidth,
          padding: 10,
        }}
      >
        <View style={{
          flexDirection: 'row',
        }}>
          { triggering_user && <UserImage source={{uri: triggering_user.image_url}} style={{marginRight: 10}} /> }
          <View style={{
            flex: 1,
            flexDirection: 'column',
          }}>
            <HtmlSentence>
              {text}
            </HtmlSentence>
            <TimeAgo time={created_at} style={{
              color: Colors.gray,
              marginTop: 4,
            }} />
          </View>
        </View>
      </TouchableOpacity>
    )
  }
}
