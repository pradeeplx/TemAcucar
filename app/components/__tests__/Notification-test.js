import 'react-native'
import React from 'react'
import Notification from '../Notification'
import renderer from 'react-test-renderer'
import { identity } from 'lodash'

const notification = {
  read: true,
  text: 'Foo',
  triggering_user: {
    image_url: 'url',
  },
  created_at: 10E4,
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Notification notification={notification} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with different color for unread', () => {
  const tree = renderer.create(
    <Notification notification={{ ...notification, read: false }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without image', () => {
  const tree = renderer.create(
    <Notification notification={{ ...notification, triggering_user: null }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
