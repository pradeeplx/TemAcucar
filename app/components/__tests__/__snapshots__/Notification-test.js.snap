exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 0.5,
      "borderColor": "#CECED2",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <Image
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          Object {
            "marginRight": 10,
          },
        ]
      } />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Array [
                  Object {},
                  undefined,
                ],
              ]
            }>
            Foo
          </Text>
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#CECED2",
                "marginTop": 4,
              },
            ],
          ]
        }>
        Há 47 anos
      </Text>
    </View>
  </View>
</View>
`;

exports[`test renders with different color for unread 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FAF6F1",
      "borderBottomWidth": 0.5,
      "borderColor": "#CECED2",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <Image
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          Object {
            "marginRight": 10,
          },
        ]
      } />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Array [
                  Object {},
                  undefined,
                ],
              ]
            }>
            Foo
          </Text>
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#CECED2",
                "marginTop": 4,
              },
            ],
          ]
        }>
        Há 47 anos
      </Text>
    </View>
  </View>
</View>
`;

exports[`test renders without image 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 0.5,
      "borderColor": "#CECED2",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Array [
                  Object {},
                  undefined,
                ],
              ]
            }>
            Foo
          </Text>
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#CECED2",
                "marginTop": 4,
              },
            ],
          ]
        }>
        Há 47 anos
      </Text>
    </View>
  </View>
</View>
`;
